### PR TITLE
mobile/ci: Adds a `mobile-test-ios` Bazel build option

### PR DESCRIPTION
**Pull request by abeyad**
Also, we rename `test-android` to `mobile-test-android` for consistency.

### CODE REVIEW

File: mobile/.bazelrc [hunk 1]
```diff
@@ -77,6 +77,12 @@ build:mobile-dbg-ios --config=ios
 # https://github.com/envoyproxy/envoy/issues/24977
 build:test-android --define=static_extension_registration=enabled
 
+# Default flags for iOS tests.
+build:test-ios --config=ios
```
Comment (phlax) [inline mobile/.bazelrc:81]: can we prefix these flags with `mobile-` please

in this case its unlikely to matter, but previously it took a while to unpick ~clashing flags with the base envoy bazelrc - which is why i added the prefix to all in this file - so they cant clash

Comment (phlax) [reply mobile/.bazelrc:81]: (just noticing that ios doesnt currently have the prefix, but i think it should)

Comment (abeyad) [reply mobile/.bazelrc]: done

**Review by phlax — DISMISSED** [review]
lgtm, thanks @abeyad

**Comment by abeyad** [comment]
We are going a different route to enable the envoy_mobile_listener factories required, so changing this PR to just add mobile-test-ios for consistency, and change test-android to mobile-test-android for consistency.

**Pull request merged**